packages/orama/src/components/sorter.ts

```diff
@@ -0,0 +1,267 @@
+    sorts: {}
+  }
+
+  for (const [prop, type] of Object.entries(schema)) {
+    const typeActualType = typeof type
+    const path = `${prefix}${prefix ? '.' : ''}${prop}`
+
+    if (sortableDeniedProperties.includes(path)) {
+      continue
+    }
+
+    if (typeActualType === 'object' && !Array.isArray(type)) {
+      // Nested
+      const ret = innerCreate(orama, type as Schema, sortableDeniedProperties, path)
+      sorter.sortableProperties.push(...ret.sortableProperties)
+      sorter.sorts = {
+        ...sorter.sorts,
+        ...ret.sorts
+      }
+      sorter.sortablePropertiesWithTypes = {
+        ...sorter.sortablePropertiesWithTypes,
+        ...ret.sortablePropertiesWithTypes
+      }
+      continue
+    }
+
+    switch (type) {
+      case 'number':
+      case 'string':
+        sorter.sortableProperties.push(path)
+        sorter.sortablePropertiesWithTypes[path] = type
+        sorter.sorts[path] = {
+          docs: {},
+          orderedDocs: [],
+          type: type,
+          n: 0
+        }
+        break
+      case 'boolean':
+      case 'boolean[]':
+      case 'number[]':
+      case 'string[]':
+        // We don't allow to sort by arrays
+        continue
+      default:
+        throw createError('INVALID_SORT_SCHEMA_TYPE', Array.isArray(type) ? 'array' : type as unknown as string, path)
+    }
+  }
+
+  return sorter
+}
+
+async function create(
+  orama: Orama<{ Sorter: Sorter }>,
+  schema: Schema,
+  config?: SorterConfig,
+): Promise<Sorter> {
+  const isSortEnabled = config?.enabled !== false
+  if (!isSortEnabled) {
+    return {
+      disabled: true
+    } as unknown as Sorter
+  }
+  return innerCreate(orama, schema, (config || {}).deniedProperties || [], '')
+}
+
+function stringSort(value: SortValue, language: string | undefined, d: [string, SortValue]): boolean {
+  return (d[1] as string).localeCompare(value as string, language) > 0
+}
+function numerSort(value: SortValue, d: [string, SortValue]): boolean {
+  return (d[1] as number) > (value as number)
+}
+
+async function insert(
+  sorter: Sorter,
+  prop: string,
+  id: string,
+  value: SortValue,
+  schemaType: SortType,
+  language: string | undefined,
+): Promise<void> {
+  if (!sorter.enabled) {
+    return
+  }
+  const s = sorter.sorts[prop] as PropertySort<SortValue>
+
+  let predicate: (value: [string, SortValue]) => boolean
+  switch(schemaType) {
+    case "string":
+      predicate = stringSort.bind(null, value, language)
+      break;
+    case "number":
+      predicate = numerSort.bind(null, value)
+      break;
+  }
+
+  // Find the right position to insert the element
+  let index = s.orderedDocs.findIndex(predicate)
+  if (index === -1) {
+    index = s.orderedDocs.length
+    s.orderedDocs.push([id, value])
+  } else {
+    s.orderedDocs.splice(index, 0, [id, value])
+  }
+  s.docs[id] = index
+
+  // Increment position for the greather documents
+  for (let i = index + 1; i < s.orderedDocs.length; i++) {
```

Comment:
Here you are accessing a nested property and calculating its length for each iteration; move this to a separate constant:

```js
const orderedDocsLength = s.orderedDocs.length
for (let i = index + 1; i < orderedDocsLength; i++) {
```